### PR TITLE
[MM-34523] New trial card for license page

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
This adds a new card for the license card.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34523
https://www.figma.com/file/QzIJRwPZVEmFhkUznuH16x/Final-Spec---EE-In-Product-Discovery?node-id=260%3A887

#### Screenshots
Screen Shot 2021-04-06 at 5 25 58 PM

### CODE REVIEW

File: components/admin_console/license_settings/trial_license_card/trial_license_card.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    const message = () => {
+        if (hoursToEndLicense > 24) {
+            return (
+                <FormattedMessage
+                    id='admin.license.trialCard.description'
+                    defaultMessage='Your free trial will expire in <b>{daysCount} {daysCount, plural, one {day} other {days}}</b>. Visit our customer portal to purchase a license now to continue using E10 & E20 features after trial ends.'
+                    values={{
+                        daysCount: daysToEndLicense,
+                        b: (chunk: any) => (<b>{chunk}</b>),
```
Comment (hahmadia) [inline components/admin_console/license_settings/trial_license_card/trial_license_card.tsx:40]: Not a huge fan of the way we are bolding text. I think it's better to use `FormattedMarkdownMessage` and use `**` around the text you want bolded instead. Avoids this extra function passing.

File: components/admin_console/license_settings/trial_license_card/trial_license_card.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            );
+        }
+
+        return (
+            <FormattedMessage
+                id='admin.license.trialCard.description.expiringToday'
+                defaultMessage='Your free trial expires <b>Today at {time}</b>. Visit our customer portal to purchase a license now to continue using E10 & E20 features after trial ends'
+                values={{
+                    time: endOfLicense.format('h:mm a') + moment().tz(getBrowserTimezone()).format('z'),
+                    b: (chunk: any) => (<b>{chunk}</b>),
```
Comment (hahmadia) [inline components/admin_console/license_settings/trial_license_card/trial_license_card.tsx:52]: Lets use formattedmarkdownmessage

File: components/announcement_bar/purchase_now_link/purchase_now_link.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        <button
+            className={'annnouncementBar__purchaseNow'}
+            onClick={(e) => handlePurchaseLinkClick(e)}
+        >
+            {props.buttonTextElement}
+        </button>
+    );
+};
+
+export default PurchaseNowLink;
```
Comment (hahmadia) [inline components/announcement_bar/purchase_now_link/purchase_now_link.tsx:29]: Thanks for the help on creating this component <3

File: utils/license_utils.jsx [hunk 4]
```diff
@@ -30,3 +30,18 @@ export function isLicensePastGracePeriod(license) {
     const timeDiff = Date.now() - parseInt(license.ExpiresAt, 10);
     return timeDiff > LICENSE_GRACE_PERIOD;
 }
+
+export function isTrialLicense(license) {
```
Comment (hahmadia) [inline utils/license_utils.jsx:34]: Nice! I used this in my code too :)

**Review by hahmadia — COMMENTED** [review]
Non blocking suggestions. Good job! 💯 🔥

File: components/announcement_bar/purchase_now_link/purchase_now_link.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import React from 'react';
+
+import {trackEvent} from 'actions/telemetry_actions';
+
+export interface Props {
+    buttonTextElement: JSX.Element;
+}
+
+const PurchaseNowLink: React.FC<Props> = (props: Props) => {
```
Comment (sbishel) [inline components/announcement_bar/purchase_now_link/purchase_now_link.tsx:12]: Would prefer to simplify name.
```suggestion
const PurchaseLink: React.FC<Props> = (props: Props) => {
```

**Review by sbishel — COMMENTED** [review]
The Formatting changes are more important than the naming changes.

**Review by hahmadia — APPROVED** [review]
LGTM now. 
I just want to mention that for the trackeEvent and telemetry, do we need to follow with the telemetry handbook Harshil linked too? I know he mentioned in the enterprise channel and linked this handbook

[Removed link for confidential reasons]

**Comment by BenCookie95** [comment]
> LGTM now.
> I just want to mention that for the trackeEvent and telemetry, do we need to follow with the telemetry handbook Harshil linked too? I know he mentioned in the enterprise channel and linked this handbook
> 
> [Removed link for confidential reasons]

I had a read through the handbook and from what I understood for the webapp all we have to do is called `trackEvent` and notify Adam of what ID to look out for? trackEvent can accept any payload

**Review by sbishel — APPROVED** [review]

**Review by furqanmlk — COMMENTED** [review]
Should there be 29 days or 30 days ?
image

**Comment by furqanmlk** [comment]
Everything looks good except few UI mismatched with Figma

image

**Comment by BenCookie95** [comment]
@furqanmlk 1 & 3 are now fixed. 2 was a typo in the figma

**Review by furqanmlk — APPROVED** [review]
Thanks @BenCookie95 
All look good, Test case have been added to test plan

**Pull request merged**